src/plots/cartesian/layout_attributes.js

```diff
@@ -279,22 +279,22 @@ module.exports = {
         pattern: {
             valType: 'enumerated',
             // TODO could add '%H:%M:%S'
-            values: ['%w', '%H', ''],
+            values: ['day of week', 'hour', ''],
             dflt: '',
             role: 'info',
             editType: 'calc',
             description: [
                 'Determines a pattern on the time line that generates breaks.',
-                'If *%w* - Sunday-based weekday as a decimal number [0, 6].',
-                'If *%H* - hour (24-hour clock) as a decimal number [0, 23].',
-                'These are the same directive as in `tickformat`, see',
-                'https://github.com/d3/d3-time-format#locale_format',
+                'If *day of week* - Sunday-based weekday as a decimal number [0, 6].',
+                'If *hour* - hour (24-hour clock) as a decimal number [0, 23].',
```

Comment:
this accepts numbers from 0 to 23.999999 right?

Reply:
Not sure.
cc: @alexcjohnson

Reply:
let's check in on the time-strings here too plz. I don't mind if they're not supported but if they are we should leave it in and document it. :)

Reply:
Fractional hours worked in my testing, and time strings did not, but the gold standard is adding a test :)

Reply:
@archmoj your call whether to add these tests here or in another PR but they should be somewhere before the release.

Reply:
I am adding them in this PR.

Reply:
Tests are added in https://github.com/plotly/plotly.js/pull/4653/commits/8cc498ecd95a17b50967fd75913ddc86e1a47a37.
At the moment this only works properly for integers [0, 24] as mentioned in https://github.com/plotly/plotly.js/pull/4653/commits/db38ce9d38ede3d4a99b1da3ecd73f631833894e.

Reply:
[demo for the bug](https://codepen.io/MojtabaSamimi/pen/MWwXeXr?editors=0010)

Reply:
Fixed in https://github.com/plotly/plotly.js/pull/4653/commits/a333acba5da962b1b442dd4e34a7203b1081616e.